pf9/modules/analytics_utils.py

```diff
@@ -0,0 +1,105 @@
+        self.ctx.params["segment_step_count"] = 1
+
+    def send_track(self, step_name):
+        """ Builds event name based on subcommand and step count
+        """
+        event_name = " - ".join([self.ctx.params["segment_event_prefix"], self.ctx.params["segment_subcommand"],
+                              step_name, str(self.ctx.params["segment_step_count"])])
+        print("Dynamic event name: {}".format(event_name))
+        self.ctx.params["segment_step_count"] = self.ctx.params["segment_step_count"] + 1
+        self.ctx.params["segment_session"].send_track(event_name, self.ctx.params["segment_event_properties"], user_id=self.ctx.params['user_id'])
+
+    def send_track_error(self, step_name, error_message):
+        error_event_name = " - ".join([self.ctx.params["segment_event_prefix"], self.ctx.params["segment_subcommand"],
+                                 step_name, "ERROR"])
+        print("Error Dynamic event name: {}".format(error_event_name))
+        print("Error Details : {}".format(error_message))
+        segment_properties = self.ctx.params["segment_event_properties"].copy()
+        segment_properties.update(dict(error_message=error_message))
+        self.ctx.params["segment_session"].send_track(error_event_name, segment_properties,
+                                                      user_id=self.ctx.params['user_id'])
+
+
+class SegmentSession:
+
+    def __init__(self):
+        # unique for the device
+        self.device_id = str(uuid.uuid3(uuid.NAMESPACE_DNS, str(uuid.getnode())))
+
+        # unique for this session
+        self.anonymousId = str(uuid.uuid1())
+
+        # Session time is time in EPOC that must be passed to Segment in each track event to ensure that a single session is maintained. The client side CLI should generate this and submit the string with each API submission
+        self.session_time = math.floor(time.time())
+
+
+
+    def send_track(self, event_name, event_properties, user_id=None):
+    # wizard_progress: started, in-progress or ended
+    # event_name: WZ PF9Express - KubeCTL 6
+    # wizard_step: 20
+        track_dict = {
+            'anonymous_id': self.anonymousId,
+            'wizard_name': event_name,
+            # 'wizard_state': wizard_state,
+            'installation_id': self.device_id,
+            'deployment_type': 'BareOS'
+        }
+        # append event_properties to track_dict as is
+        track_dict.update(event_properties)
+        track_user_id = self.anonymousId
+        if user_id:
+            track_dict['user_id'] = user_id
+            track_user_id = user_id
+
+        analytics.track(track_user_id, event_name, track_dict,
+            anonymous_id=self.anonymousId,
+            # The 'integrations array begin passed below is how the 'session' identifier is passed into Amplitude
+            integrations={
+                'Amplitude': {
+                    'session_id': self.session_time
+                }
```

Comment:
The purpose of segment was to avoid amplitude specific code in the CLI. Looks like that is not entirely true.

Reply:
Looks like this is needed to have a notion of session, it is a amplitude specific notion and segment is not aware of it. 
Ref: https://segment.com/docs/connections/destinations/catalog/amplitude/#sessionid